### PR TITLE
Remove the flags what to expect WASI targets.

### PR DESCRIPTION
**Pull request by langyo**
cc https://github.com/rustwasm/gloo/pull/411#discussion_r1421370667 https://github.com/rustwasm/gloo/pull/424#issuecomment-1848357604
@futursolo @hamza1311

### CODE REVIEW

**Comment by langyo** [comment]
The process of removing `#[cfg(not(target_os = "wasi"))]` is not going smoothly. I will temporarily mark this PR as a draft...

Previously, I had to strip off the `History` object that cannot be used normally in WASI. Because all imports related to `web_sys` will cause problems under the `wasm32-wasi` compilation target.

File: crates/history/Cargo.toml [hunk 1]
```diff
@@ -18,11 +18,9 @@ serde = { version = "1", features = ["derive"] }
 serde-wasm-bindgen = "0.6.0"
 serde_urlencoded = { version = "0.7", optional = true }
 thiserror = { version = "1.0", optional = true }
-
-[target.'cfg(not(target_os = "wasi"))'.dependencies]
 wasm-bindgen = "0.2"
```
Comment (futursolo) [inline crates/history/Cargo.toml:21]:
```suggestion
wasm-bindgen = "0.2.88"
```

**Review by futursolo — COMMENTED** [review]
I originally intended this to be fixed in #423, could you please update the wasm-bindgen version here?

**Review by futursolo — APPROVED** [review]
Thank you. Looks good to me.

**Comment by langyo** [comment]
@hamza1311 Now `gloo-history` should be able to release a patch version... Once the patch version is released, `yew` can officially start review. Could you please take a look?

@futursolo And could you please help me see if this can be merged as soon as possible? https://github.com/yewstack/prokio/pull/11

> The original PR is here https://github.com/yewstack/yew/pull/3534

**Comment by hamza1311** [comment]
Published gloo-history v0.2.2 :tada:

**Pull request merged**